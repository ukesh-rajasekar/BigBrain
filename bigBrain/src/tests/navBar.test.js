import React from 'react'
import { fireEvent, render } from '@testing-library/react';
import { configure } from 'enzyme';
import Navbar from '../components/navBar';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });



describe('Render NvaBar', () => {

    test('Go to Dashboard link is there', () => {
        const NavComponent = render(<Navbar/>)
        const text = NavComponent.getByText('Dashboard')
        expect(text).toBeInTheDocument;
    })

    test('dashboard routing works', () => {
        const NavComponent = render(<Navbar/>)
        const dashboardRoute = NavComponent.getByText('Dashboard');
        expect(dashboardRoute.getAttribute("href")).toBe(
            "/admin")
            
    });

    test('Go to Games link is there', () => {
        const NavComponent = render(<Navbar/>)
        const text = NavComponent.getByText('Games')
        expect(text).toBeInTheDocument;
    })

    test('Games routing works', () => {
        const NavComponent = render(<Navbar/>)
        const gameRoute = NavComponent.getByText('Games');
        expect(gameRoute.getAttribute("href")).toBe(
            "/Home")
            
    });

    test('Go to UploadGame link is there', () => {
        const NavComponent = render(<Navbar/>)
        const text = NavComponent.getByText('Upload Game')
        expect(text).toBeInTheDocument;
    })

    test('Upload Game routing works', () => {
        const NavComponent = render(<Navbar/>)
        const uploadGameRoute = NavComponent.getByText('Upload Game');
        expect(uploadGameRoute.getAttribute("href")).toBe(
            "admin/uploadGame")
            
    });

    test('Log Out button is there', () => {
        const NavComponent = render(<Navbar/>)
        const text = NavComponent.getByText('Log Out')
        expect(text).toBeInTheDocument;
    })

      test('render with min props', () => {  
        const NavComponent = renderer.create(<Navbar/>).toJSON();
        expect(NavComponent).toMatchSnapshot();
      });
  });
