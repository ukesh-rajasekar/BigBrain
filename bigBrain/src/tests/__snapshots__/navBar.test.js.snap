// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render NvaBar render with min props 1`] = `
<nav
  className="navbar navbar-expand navbar-dark bg-primary"
>
  <a
    className="navbar-brand"
    href="/admin"
  >
    Dashboard 
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-home fa-w-18 "
      data-icon="home"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 576 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </a>
  <div
    className="mr-auto navbar-nav"
    onKeyDown={[Function]}
  >
    <a
      className="nav-link"
      data-rb-event-key="admin/uploadGame"
      href="admin/uploadGame"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      Upload Game 
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-upload fa-w-16 "
        data-icon="upload"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M296 384h-80c-13.3 0-24-10.7-24-24V192h-87.7c-17.8 0-26.7-21.5-14.1-34.1L242.3 5.7c7.5-7.5 19.8-7.5 27.3 0l152.2 152.2c12.6 12.6 3.7 34.1-14.1 34.1H320v168c0 13.3-10.7 24-24 24zm216-8v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h136v8c0 30.9 25.1 56 56 56h80c30.9 0 56-25.1 56-56v-8h136c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </a>
  </div>
  <button
    className="btn btn-primary btn-sm"
    disabled={false}
    name="logout"
    onClick={[Function]}
    type="button"
  >
    Log Out 
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-sign-out-alt fa-w-16 "
      color="white"
      data-icon="sign-out-alt"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
      
  </button>
</nav>
`;
